src/compiler/checker.ts

```diff
@@ -48501,11 +48249,36 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     function isLiteralConstDeclaration(node: VariableDeclaration | PropertyDeclaration | PropertySignature | ParameterDeclaration): boolean {
-        if (isDeclarationReadonly(node) || isVariableDeclaration(node) && isVarConstLike(node)) {
+        if (isDeclarationReadonly(node) || (isVariableDeclaration(node) && isVarConstLike(node)) || isEnumMember(node)) {
+            if (compilerOptions.isolatedDeclarations) {
+                return !!node.initializer && !node.type && isPrimitiveLiteralValue(node.initializer);
+            }
             return isFreshLiteralType(getTypeOfSymbol(getSymbolOfDeclaration(node)));
         }
         return false;
     }
+    function isLiteralComputedName(node: ComputedPropertyName) {

```

Comment:
I'm less familiar with computed property handling, but this code must also already exist in the checker in order for it to work. So I expected to see some use of it inside the checker instead of just exporting it.